inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/ConceptInstancePanel.java

```diff
@@ -134,17 +136,31 @@ public void actionStatementChanged(AjaxStatementChangedEvent event)
             return;
         }
 
-        boolean isRenameEvent = statement.getProperty().getIdentifier()
-                .equals(RDFS.LABEL.stringValue());
-        if (!isRenameEvent) {
+        if (!isRenamingEvent(statement)) {
             return;
         }
-
-        instanceHandle.setName((String) statement.getValue());
+        Optional<KBInstance> kbInstance = kbService
+            .readInstance(kbModel.getObject(),
+                statement.getInstance().getIdentifier());
+        if (kbInstance.isPresent()) {
+            instanceHandle.setName(kbInstance.get().getName());
+        }
         selectedInstanceHandle.setObject(instanceHandle);
         event.getTarget().add(this);
     }
 
+    private boolean isRenamingEvent(KBStatement aStatement)
+    {
```

Comment:
Can you please add JavaDoc here - what is this method doing? The argument is a statement, not an event - so I don't understand how you can check if the statement is a renaming event.

Reply:
No idea on that. @MdeBoer95 what was the logic behind this method and do we still need that?

Reply:
It determines whether the event is about renaming a specific Instance. So yes it would make more sense to pass an event as a parameter here and then get the statement from that event.
I think we still need that because without it changing the label might not update the Instance name at the top of the Panel. 
But the ``isSubpropertyLabel`` method in the ``kbService`` needs to be updated since it doesn't take transitive subproperties into account right now.

Reply:
Ok. Yeah. I would need to do some changes here.

Reply:
@MdeBoer95 can u please guide me to why exactly do we need this in context of sub property labels here. Also,  can you please add a java doc to the method here.

Reply:
Whenever a label of a selected instance is changed we want to update the ``selectedInstanceHandle``object. So each time a statement is changed, we need to check whether this change involves the label of the selected instance. Since the label could now also be a subproperty label and not only a direct label like ``rdfs:label``, we need to adjust the check method (``isRenamingEvent()``)  to take subproperty labels into account .